server/src/main/java/org/opensearch/tasks/SearchShardTaskCancellationStats.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.tasks;
+
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class SearchShardTaskCancellationStats implements ToXContentObject, Writeable {
+
+    private final long currentLongRunningCancelledTaskCount;
+    private final long totalLongRunningCancelledTaskCount;
```

Comment:
We should also track the count of total cancelled tasks, irrespective of it breaching the long running threshold. That also ensures we are not leaking any entries from the map

Reply:
Missed this. But I think to ensure no leaking from the map, it is better to add more tests than adding count of cancelled tasks. But if we still feel a need, we can take it up later.

Reply:
Created an issue to track this for later - https://github.com/opensearch-project/OpenSearch/issues/7977